packages/nx/src/project-graph/build-dependencies/explicit-project-dependencies.spec.ts

```diff
@@ -1,223 +1,473 @@
+
+      expect(res).toEqual([
+        {
+          sourceProjectName,
+          sourceProjectFile: 'libs/proj/file-1.ts',
+          targetProjectName: 'proj4ab',
         },
-      },
-    };
-    const nxJson = {
-      npmScope: 'proj',
-    };
-    const tsConfig = {
-      compilerOptions: {
-        baseUrl: '.',
-        paths: {
-          '@proj/proj': ['libs/proj/index.ts'],
-          '@proj/my-second-proj': ['libs/proj2/index.ts'],
-          '@proj/project-3': ['libs/proj3a/index.ts'],
-          '@proj/proj123': ['libs/proj123/index.ts'],
-          '@proj/proj1234': ['libs/proj1234/index.ts'],
-          '@proj/proj1234-child': ['libs/proj1234-child/index.ts'],
-          '@proj/proj4ab': ['libs/proj4ab/index.ts'],
+        {
+          sourceProjectName,
+          sourceProjectFile: 'libs/proj/file-2.ts',
+          targetProjectName: 'proj3a',
         },
+      ]);
+    });
+
+    it('should not build explicit dependencies when nx-ignore-next-line comments are present', () => {
+      const sourceProjectName = 'proj';
+      const { ctx, builder } = createFixture({
+        sourceProjectName,
+        sourceProjectFiles: [
+          {
+            path: 'libs/proj/file-1.ts',
+            content: `
+              const a = {
+                // nx-ignore-next-line
+                loadChildren: '@proj/proj4ab#a'
+              };
+            `,
+          },
+          /**
+           * TODO: This case, where a multi-line comment is used, is not working
+           */
+          // {
+          //   path: 'libs/proj/file-2.ts',
+          //   content: `
+          //     const a = {
+          //       /* nx-ignore-next-line */
+          //       loadChildren: '@proj/proj4ab#a'
+          //     };
+          //   `,
+          // },
+          /**
+           * TODO: These cases, where loadChildren is on the same line as the variable declaration, are not working
+           */
+          // {
+          //   path: 'libs/proj/file-3.ts',
+          //   content: `
+          //     // nx-ignore-next-line
+          //     const a = { loadChildren: '@proj/proj4ab#a' };
+          //   `,
+          // },
+          // {
+          //   path: 'libs/proj/file-4.ts',
+          //   content: `
+          //     /* nx-ignore-next-line */
+          //     const a = { loadChildren: '@proj/proj4ab#a' };
+          //   `,
+          // },
+        ],
+      });
+
+      const res = buildExplicitTypeScriptDependencies(
+        ctx.workspace,
+        builder.graph,
+        ctx.filesToProcess
+      );
+
+      expect(res).toEqual([]);
+    });
+  });
+});
+
+interface Fixture {
+  sourceProjectName: string;
+  sourceProjectFiles: {
+    path: string;
+    content: string;
+  }[];
+}
+
+/**
+ * Prepares a minimal workspace and virtual file-system for the given files and dependency
+ * projects in order to be able to execute `buildExplicitTypeScriptDependencies()` in the tests.
+ */
+function createFixture(fixture: Fixture) {
```

Comment:
Abstractions in unit tests should always be approached cautiously, but in this particular case because of the sheer amount of setup required, and the fact that the abstraction is entirely local to this spec file, I feel that this is worth it for the great enhancement to readability of the cases above that it provides

Reply:
(Renamed to `createVirtualWorkspace` to avoid ambiguity with things on disk, per Victor's request)